src/core/tests/unittests/resize-test.cpp

```diff
@@ -0,0 +1,215 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2018-2019 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+
+#include <warn/push>
+#include <warn/ignore/all>
+#include <gtest/gtest.h>
+#include <warn/pop>
+
+#include <inviwo/core/common/inviwo.h>
+#include <inviwo/core/common/inviwoapplication.h>
+
+#include <inviwo/core/processors/processor.h>
+#include <inviwo/core/network/processornetwork.h>
+#include <inviwo/core/network/processornetworkevaluator.h>
+#include <inviwo/core/network/networklock.h>
+
+#include <inviwo/core/ports/imageport.h>
+#include <inviwo/core/interaction/events/resizeevent.h>
+
+#include <functional>
+
+namespace inviwo {
+
+namespace {
+
+struct TestProcessor : Processor {
+    TestProcessor(const std::string& id) : Processor(id, id), inport("inport"), outport("outport") {
+        addPort(inport);
+        addPort(outport);
+    }
+
+    virtual const ProcessorInfo getProcessorInfo() const override { return processorInfo_; }
+
+    static const ProcessorInfo processorInfo_;
+
+    virtual void initializeResources() override {}
+    virtual void process() override {}
+    virtual void doIfNotReady() override {}
+
+    ImageInport inport;
+    ImageOutport outport;
+};
+
+// The Class Identifier has to be globally unique. Use a reverse DNS naming scheme
+const ProcessorInfo TestProcessor::processorInfo_{
+    "org.inviwo.TestProcessor",  // Class identifier
```

Comment:
`"org.inviwo.TestProcessor"` is probably not as unique. Consider `"org.inviwo.gtest.TestProcessor"`

Reply:
does not matter really since it is never registered anywhere..

Reply:
OK